src/api/session/ui/Imp-qt.cpp

```diff
@@ -261,6 +261,28 @@ auto ImpQt::BlockchainSelectionQt(
     return it->second.get();
 }
 
+auto ImpQt::BlockchainSelectionQt(
+        const opentxs::ui::Blockchains key,
+        const std::vector<opentxs::blockchain::Type>& chains,
+        const SimpleCallback updateCB) const noexcept
+    -> opentxs::ui::BlockchainSelectionQt*
+{
+    auto lock = Lock{lock_};
+
+    auto& native = blockchain_selection(lock, key, chains, updateCB);
+    auto it = blockchain_selection_qt_
+                .emplace(
+                    key,
+                    std::make_unique<opentxs::ui::BlockchainSelectionQt>(
+                        *native))
+                .first;
```

Comment:
you can use try_emplace to avoid creating stuff on heap in case of having key in map :P